core/include/traccc/clusterization/impl/sparse_ccl.ipp

```diff
@@ -41,23 +41,27 @@ TRACCC_HOST_DEVICE inline unsigned int make_union(
     return e;
 }
 
-TRACCC_HOST_DEVICE inline bool is_adjacent(const traccc::cell& a,
-                                           const traccc::cell& b) {
+template <typename T1, typename T2>
+TRACCC_HOST_DEVICE inline bool is_adjacent(const edm::silicon_cell<T1>& a,
+                                           const edm::silicon_cell<T2>& b) {
 
-    return (a.channel0 - b.channel0) * (a.channel0 - b.channel0) <= 1 &&
-           (a.channel1 - b.channel1) * (a.channel1 - b.channel1) <= 1 &&
-           a.module_link == b.module_link;
+    return (a.channel0() - b.channel0()) * (a.channel0() - b.channel0()) <= 1 &&
+           (a.channel1() - b.channel1()) * (a.channel1() - b.channel1()) <= 1 &&
+           a.module_index() == b.module_index();
 }
 
-TRACCC_HOST_DEVICE inline bool is_far_enough(const traccc::cell& a,
-                                             const traccc::cell& b) {
+template <typename T1, typename T2>
+TRACCC_HOST_DEVICE inline bool is_far_enough(const edm::silicon_cell<T1>& a,
+                                             const edm::silicon_cell<T2>& b) {
 
-    assert((a.channel1 >= b.channel1) || (a.module_link != b.module_link));
-    return (a.channel1 > (b.channel1 + 1)) || (a.module_link != b.module_link);
+    assert((a.channel1() >= b.channel1()) ||
+           (a.module_index() != b.module_index()));
+    return (a.channel1() > (b.channel1() + 1)) ||
+           (a.module_index() != b.module_index());
```

Comment:
Same story again, these should be universal references to cell-like objects.

Reply:
You made the comment on the same function (the declaration and the implementation) twice...?